cranelift/codegen/src/isa/zkasm/inst/mod.rs

```diff
@@ -1250,6 +1250,64 @@ impl Inst {
             &Inst::ECall {} => String::from("ecall"),
         }
     }
+
+    fn print_name(&self) -> String {
+        let name = match self {
+            &Inst::Nop => "Nop",
```

Comment:
This is kinda neat, but I think we should also consider reusing this for `print_with_state` to maintain consistent names across various places where we print these instructions. This shouldn't be too hard (although the diff will not be minimal) and can be achieved roughly through:

```rust
let mut output = String::with_capacity(32);
output.push_str(self.print_name);
match self {
    Inst::Udf { trap_code } => write!(&mut output, "##trap_code={trap_code}"),
    ...
}
```

This should also reveal that e.g. `Inst::Select` is actually considered multiple different instructions depending on a `ty` field. It might or might not be useful to split them out for histogram purposes as well.